public/docs/dart/latest/guide/forms.jade

```diff
@@ -0,0 +1,662 @@
+  1. Show and hide validation error messages.
+  1. Handle form submission with **ng-submit**.
+  1. Change the form's display after submission.
+
+:marked
+  ## Setup
+  Create a new project folder (`angular2_forms`) and create 3 files:
+  `pubspec.yaml`, `web/index.html`, and `web/main.dart`.
+  (These files should be familiar from the
+  [QuickStart](../quickstart.html).) Put these contents in the files:
+
++makeTabs('forms/dart/angular2_forms-1-SETUP/pubspec.yaml, forms/dart/angular2_forms-1-SETUP/web/index.html, forms/dart/angular2_forms-1-SETUP/web/main.dart', null, 'pubspec.yaml, web/index.html, web/main.dart')
+
+:marked
+  So that the code can run,
+  let's create a stub for the `<hero-form>` component.
+
+  Create a new directory called `lib`.
+  In it, put a file called `hero_form_component.dart`
+  with the following code:
+
++makeExample('forms/dart/angular2_forms-1-SETUP/lib/hero_form_component.dart', null, 'lib/hero_form_component.dart')
+
+:marked
+  The app should now run, but it won't do anything interesting.
+  Let's add some data.
+
+
+  ## Create the Hero model class
+
+  As users enter form data, we'll capture their changes and update an instance of a model.
+  We can't lay out the form until we know what the model looks like.
+
+  A model can be as simple as a "property bag" that holds facts about a thing of application importance.
+  That describes well our `Hero` class with its three required fields (`id`, `name`, `power`)
+  and one optional field (`alterEgo`).
+
+  In the `lib` directory, add a file called `hero.dart`
+  with the following code:
+
++makeExample('forms/dart/angular2_forms-2-COMPONENT/lib/hero.dart', 'all', 'lib/hero.dart')
+
+:marked
+  It's an anemic model with few requirements and no behavior. Perfect for our demo.
+
+  Here's an example of creating and using a Hero object:
+
++makeExample('forms/dart/angular2_forms-2-COMPONENT/lib/hero.dart', 'newhero')(format=".")
+
+:marked
+  The `alterEgo` is optional, so the constructor lets us omit it: note the
+  `[]` in `[this.alterEgo]`.
+
+
+.l-main-section
+:marked
+  ## Create a form component
+
+  An Angular form has two parts: an HTML-based template and a code-based component to handle data and user interactions.
+
+  We begin with the component because it states, in brief, what the Hero editor can do.
+
+  Edit `hero_form_component.dart`, replacing all of its contents
+  with the following code:
+
++makeExample('forms/dart/angular2_forms-2-COMPONENT/lib/hero_form_component.dart', null, 'lib/hero_form_component.dart')
+
+:marked
+  There’s nothing special about this component, nothing to distinguish it from any component we've written before,
+  nothing form-specific about it ... except, perhaps, the tell-tale `FORM_DIRECTIVES` import.
+
+  Understanding this component requires only the Angular 2 concepts covered in previous guides.
+
+  1. The code imports a standard set of symbols from the Angular library.
```

Comment:
We're going to have to sweep ALL documents and pull out every place that we add `FORM_DIRECTIVES` and `CORE_DIRECTIVES`.  A PITA for us but good for developers.